packages/contracts-bedrock/src/L1/BatchVerifier.sol

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.28;
+
+import {ECDSA} from "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import {OwnableUpgradeable} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import {ISemver} from "interfaces/universal/ISemver.sol";
+import {IEspressoNitroTEEVerifier} from "@espresso-tee-contracts/interface/IEspressoNitroTEEVerifier.sol";
+import {EspressoTEEVerifier} from "@espresso-tee-contracts/EspressoTEEVerifier.sol";
+import {IEspressoTEEVerifier} from "@espresso-tee-contracts/interface/IEspressoTEEVerifier.sol";
+
+interface INitroValidator {
+    function decodeAttestationTbs(
+        bytes memory attestation
+    )
+        external
+        pure
+        returns (bytes memory attestationTbs, bytes memory signature);
+}
+
+contract BatchVerifier is ISemver, OwnableUpgradeable {
```

Comment:
Would not it be easier to allow the `BatchInbox` contract to update the address of the `BatchVerifier` contract instead of making the `BatchVerifier` contract upgradeable?

Reply:
We can't have any functions other than `fallback` in `BatchInbox` in our design, this keeps changes to derivation pipeline small. Otherwise we'd need to change batch format.